source/hash_table.c

```diff
@@ -235,14 +237,16 @@ void aws_hash_table_clean_up(struct aws_hash_table *map) {
 }
 
 void aws_hash_table_swap(struct aws_hash_table *AWS_RESTRICT a, struct aws_hash_table *AWS_RESTRICT b) {
-    AWS_PRECONDITION(a != b);
+    AWS_PRECONDITION(a != b, "Input hash_tables [a] and [b] should be different.");
     struct aws_hash_table tmp = *a;
     *a = *b;
     *b = tmp;
 }
 
 void aws_hash_table_move(struct aws_hash_table *AWS_RESTRICT to, struct aws_hash_table *AWS_RESTRICT from) {
-    AWS_PRECONDITION(to != NULL && from != NULL && to != from && aws_hash_table_is_valid(from));
+    AWS_PRECONDITION(
+        to && from && to != from && aws_hash_table_is_valid(from),
+        "Input hash tables [to] and [from] should be different and [from] should also be valid.");
```

Comment:
Same as above.  Might be cleaner to sepearte this into several conditions.

Reply:
I agree for both the above.